pkg/apis/volumesnapshot/v1alpha1/types.go

```diff
@@ -0,0 +1,338 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"encoding/json"
+
+	core_v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	// VolumeSnapshotDataResourcePlural is "volumesnapshotdatas"
+	VolumeSnapshotDataResourcePlural = "volumesnapshotdatas"
+	// VolumeSnapshotResourcePlural is "volumesnapshots"
+	VolumeSnapshotResourcePlural = "volumesnapshots"
+	// SnapshotClassResourcePlural is "snapshotclasss"
+	SnapshotClassResourcePlural = "snapshotclasses"
+)
+
+// VolumeSnapshotStatus is the status of the VolumeSnapshot
+type VolumeSnapshotStatus struct {
+	// The time the snapshot was successfully created
+	// +optional
+	CreationTimestamp metav1.Time `json:"creationTimestamp" protobuf:"bytes,1,opt,name=creationTimestamp"`
+
+	// Represent the latest available observations about the volume snapshot
+	Conditions []VolumeSnapshotCondition `json:"conditions" protobuf:"bytes,2,rep,name=conditions"`
+}
+
+// VolumeSnapshotConditionType is the type of VolumeSnapshot conditions
+type VolumeSnapshotConditionType string
+
+// These are valid conditions of a volume snapshot.
```

Comment:
would recommend reviewing the updated condition guidelines in https://github.com/kubernetes/community/pull/2350/files

Reply:
Changed VolumeSnapshotConditionType to "Created", "Available", and "Error" following today's discussion during the snapshot design meeting.